src/codacy/reporter.py

```diff
@@ -76,26 +75,37 @@ def strip_prefix(line, prefix):
     return filename
 
 
-def merge_reports(report_list):
-    """Merges together several report structures from parse_report_file"""
-    final_report = {
-        'language': "python",
-        'fileReports': []
-    }
+def merge_and_round_reports(report_list):
+    """Merges together several report structures from parse_report_file (and rounds all values)"""
+
+    if len(report_list) == 1:
+        final_report = report_list[0]
+    else:
+        final_report = {
+            'language': "python",
+            'fileReports': []
+        }
 
-    for report in report_list:
-        # First, merge together detailed report structures
-        # This assumes no overlap
-        # TODO: What should we do if there is a file listed multiple times?
-        final_report['fileReports'] += report['fileReports']
+        total_lines = 0
+        for report in report_list:
+            # First, merge together detailed report structures
+            # This assumes no overlap
+            # TODO: What should we do if there is a file listed multiple times?
+            final_report['fileReports'] += report['fileReports']
+            total_lines += report['codeLines']
 
-    # Gather all per-file coverage
-    total_coverages = []
-    for fileentry in final_report['fileReports']:
-        total_coverages += [fileentry['total']]
+        # Coverage weighted average (by number of lines of code) of all files
+        average_sum = 0
+        for file_entry in final_report['fileReports']:
+            average_sum += file_entry['total'] * file_entry['codeLines']
 
-    # And average
-    final_report['total'] = int(sum(total_coverages)/len(total_coverages))
+        final_report['total'] = average_sum / total_lines
+        final_report['codeLines'] = total_lines
+
+    # Round all total values
+    for file_entry in final_report['fileReports']:
+        file_entry['total'] = int(floor(file_entry['total']))
```

Comment:
I think `floor` is not needed: https://docs.python.org/2/library/functions.html#int